ci/task/compile_models.py

```diff
@@ -0,0 +1,240 @@
+    shards: int = 1
+
+    def __str__(self):
+        short_target_str = f"-{self.target.split()[0]}" if self.target else ""
+        shards_str = f"-shards-{self.shards}" if self.shards > 1 else ""
+        return f"{self.model_name}-{self.quantization}{short_target_str}{shards_str}"
+
+    def get_output_path(self) -> Path:
+        return self.output_path / f"{self.model_name}-{self.quantization}"
+
+
+def call_subprocess(cmd: Sequence[str]) -> Optional[CalledProcessError]:
+    result = subprocess.run(
+        cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, check=False, text=True
+    )
+    if result.returncode != 0:
+        return CalledProcessError(result.returncode, cmd, result.stdout)
+    return None
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--models", nargs="+", choices=MODEL_LIST, default=MODEL_LIST)
+    parser.add_argument("--local-path", type=Path, default=None)
+    parser.add_argument("--hf-token", type=str, default=None)
+    parser.add_argument(
+        "--quantization",
+        nargs="+",
+        choices=QUANTIZATION_LIST,
+        default=["q0f16", "q3f16_0", "q3f16_1", "q4f16_0", "q4f16_1", "q4f32_1"],
+    )
+    parser.add_argument("--targets", nargs="+", required=True)
+    return parser.parse_args()
+
+
+def download_model(
+    model_id: str,
+    model_path: Path,
+    hf_token: Optional[str] = None,
+    max_attempts=5,
+):
+    logging.info("Downloading model config from %s", model_id)
+    attempt = 0
+    while attempt < max_attempts:
+        try:
+            attempt += 1
+            snapshot_download(
+                repo_id=model_id,
+                local_dir=model_path,
+                local_dir_use_symlinks=False,
+                token=hf_token,
+                allow_patterns=["config.json"],  # Only download config file
+            )
+            logging.info("Download model config succeeded from %s", model_id)
+            return
+        except Exception as e:  # pylint: disable=broad-except
+            logging.warning(
+                "Download model %s config failed. Attempt number: %d", model_id, max_attempts
+            )
+            if attempt >= max_attempts:
+                raise e
+
+
+def get_target_file_name(target):
+    for pattern, value in TARGET_FILE_NAMES.items():
+        if re.search(pattern, target):
+            return value
+    raise ValueError(f"Unsupported target, expected target with regex {TARGET_FILE_NAMES.keys()}")
+
+
+def gen_config(config: CompileConfig) -> Optional[Exception]:
+    model_name = config.model_name
+    quantization = config.quantization
+    # cmd example:
+    # mlc_chat gen_config dist/models/Llama-2-7b-chat-hf/ \
+    #     --quantization q0f16 \
+    #     --conv-template test-dummy \
+    #     -o dist/libs/Llama-2-7b-chat-hf-q0f16/
+    cmd = [*MLC_CHAT, "gen_config", str(config.model_path)]
+    cmd += ["--quantization", quantization]
+    cmd += ["--conv-template", "test-dummy"]
```

Comment:
By default, use "language model" (LM), which is vanilla language model

```suggestion
    cmd += ["--conv-template", "LM"]
```